diff --git a/sdk/sdk/src/arch/linux/net_socket.cpp b/sdk/sdk/src/arch/linux/net_socket.cpp
index 0124a13..902c2dd 100644
--- a/sdk/sdk/src/arch/linux/net_socket.cpp
+++ b/sdk/sdk/src/arch/linux/net_socket.cpp
@@ -167,7 +167,7 @@ u_result SocketAddress::getAddressAsString(char * buffer, size_t buffersize) con
 
         break;
     }
-    return ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
+    return !ans?RESULT_OPERATION_FAIL:RESULT_OK;
 }
 
 
diff --git a/sdk/sdk/src/arch/win32/net_socket.cpp b/sdk/sdk/src/arch/win32/net_socket.cpp
index c2968b6..76076cb 100644
--- a/sdk/sdk/src/arch/win32/net_socket.cpp
+++ b/sdk/sdk/src/arch/win32/net_socket.cpp
@@ -244,7 +244,7 @@ u_result SocketAddress::getAddressAsString(char * buffer, size_t buffersize) con
 
         break;
     }
-    return ans<=0?RESULT_OPERATION_FAIL:RESULT_OK;
+    return !ans?RESULT_OPERATION_FAIL:RESULT_OK;
 }
 
 
